Library/Homebrew/cask/artifact/symlinked.rb

```diff
@@ -40,17 +40,25 @@ def summarize_installed
 
       private
 
-      def link(**options)
+      def link(force: false, **options)
         unless source.exist?
           raise CaskError,
                 "It seems the #{self.class.link_type_english_name.downcase} " \
                 "source '#{source}' is not there."
         end
 
         if target.exist?
-          raise CaskError,
-                "It seems there already exists #{self.class.english_article} " \
-                "#{self.class.english_name} at '#{target}'; not overwriting."
+          message = "It seems there is already #{self.class.english_article} " \
+                    "#{self.class.english_name} at '#{target}'"
+
+          raise CaskError, "#{message}." unless force
+
+          raise CaskError, "#{message}." unless target.symlink?
+
+          raise CaskError, "#{message}." unless cask.artifacts.to_s.include? target.readlink.to_s
```

Comment:
I think this could be simplified to `target.realpath == source.realpath`, i. e. a symlink already pointing to the correct location.

Reply:
Ah, much nicer -- I did not think to use `realpath`.